homeassistant/components/attribute_as_sensor/sensor.py

```diff
@@ -0,0 +1,192 @@
+"""Support for displaying attributes as Sensor."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    CONF_STATE_CLASS,
+    DEVICE_CLASSES_SCHEMA,
+    PLATFORM_SCHEMA as PARENT_PLATFORM_SCHEMA,
+    STATE_CLASSES_SCHEMA,
+    SensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_ATTRIBUTE,
+    CONF_DEVICE_CLASS,
+    CONF_ICON,
+    CONF_NAME,
+    CONF_UNIQUE_ID,
+    CONF_UNIT_OF_MEASUREMENT,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import Event, HomeAssistant, State, callback
+from homeassistant.helpers import config_validation as cv, entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.reload import async_setup_reload_service
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import PLATFORMS
+from .const import CONF_ENTITY_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PARENT_PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
```

Comment:
```suggestion
        vol.Required(CONF_SOURCE_SENSOR): cv.entity_id,
```
May be less ambiguous to use ` CONF_SOURCE_SENSOR` which is what other derived helpers like integration use https://github.com/home-assistant/core/blob/30920c3da7a714c5fef85fba4bd84191a9e5f44d/homeassistant/components/integration/sensor.py#L72

Reply:
I did actually start by using the `min_max` helper as the base e.g. copy/paste so it's not streamlined thorugh all helpers.
Also it does not neccessarily need a sensor but could be other entities so I would not agree to change this.

Reply:
OK, fair point 👍